com.dynamo.cr/com.dynamo.cr.bob.test/src/com/dynamo/bob/cache/test/ResourceCacheKeyTest.java

```diff
@@ -97,103 +101,233 @@ public void tearDown() throws IOException {
-
-		Map<String, String> ignoredOptions1 = new HashMap<String, String>();
-		ignoredOptions1.put("email", "foo@bar.com");
-		ignoredOptions1.put("debug", "true");
+		DummyBuilder.addParamsDigest(DummyBuilder.class, Map.ofEntries(
+				entry("not_important_option", "yep")
+		), DummyBuilder.class.getAnnotation(BuilderParams.class));
 		DummyBuilder builder1 = new DummyBuilder();
 		Task task1 = builder1.addInput(input).addOutput(output).create(null);
-		String key1 = ResourceCacheKey.calculate(task1, ignoredOptions1, output);
+		String key1 = ResourceCacheKey.calculate(task1.calculateSignature(), output);
 
-		Map<String, String> ignoredOptions2 = new HashMap<String, String>();
-		ignoredOptions2.put("email", "bob@acme.com");
-		ignoredOptions2.put("debug", "false");
 		DummyBuilder builder2 = new DummyBuilder();
 		Task task2 = builder2.addInput(input).addOutput(output).create(null);
-		String key2 = ResourceCacheKey.calculate(task2, ignoredOptions2, output);
+		String key2 = ResourceCacheKey.calculate(task2.calculateSignature(), output);
 
 		assertEquals(key1, key2);
 	}
+
+	// Ensure that all parameters specified in all builders exist in Bob
+	@Test
+	public void testAllParametersExist() throws IOException, CompileExceptionError, NoSuchFieldException, IllegalAccessException {
+		// Initialize project
+		Project project = new Project(new DefaultFileSystem());
+		project.scanJavaClasses();
+
+		// Access private static field classToParamsDigest
+		Class<?> builderClass = Builder.class;
+		Field field = builderClass.getDeclaredField("classToParamsDigest");
+		field.setAccessible(true);
+		Map<Class<?>, byte[]> map = (Map<Class<?>, byte[]>) field.get(null);
+
+		// Get all available command line options
+		List<Bob.CommandLineOption> commandLineOptions = Bob.getCommandLineOptions();
+		Set<String> allOptions = new HashSet<>();
+
+		// Collect all long options
+		for (Bob.CommandLineOption option : commandLineOptions) {
+			allOptions.add(option.longOpt);
+		}
+
+		// Validate each parameter in classToParamsDigest
+		for (Class<?> klass : map.keySet()) {
+			String[] params = klass.getAnnotation(BuilderParams.class).paramsForSignature();
+
+			for (String param : params) {
+				if (!allOptions.contains(param)) {
+					if (param.equals("important_option"))
+					{
+						assertEquals(BuilderParams.class.toString(), "interface com.dynamo.bob.BuilderParams");
+					}
+					else {
+						Assert.fail("Class " + klass.getName() + " uses parameter '" + param + "' in classToParamsDigest, but it does not exist in the command line options.");
+					}
+				}
+			}
+		}
+	}
+
+	// Fails if a new Bob parameter is added.
+	// Ensure that new Bob parameters do not affect build signatures.
+	// If a parameter does affect builders, add it to @BuilderParams -> paramsForSignature,
+	// then update the list in this test.
+	@Test
+	public void testNoNewParametersAdded() throws IOException, CompileExceptionError, NoSuchFieldException, IllegalAccessException {
```

Comment:
I need to make sure that every time new parameter added, the developer who added it take into account this new logic. So, I added this test to make sure it will fail and the developer will check why and take it into considiration